optimum/habana/transformers/generation/utils.py

```diff
@@ -446,6 +446,11 @@ def generate(
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
         self._validate_generated_length(generation_config, input_ids_length, has_default_max_length)
 
+        # determine whether attention softmax needs to execute in lower precision
+        model_kwargs["attn_softmax_bf16"] = (
+            generation_config.attn_softmax_bf16 if hasattr(generation_config, "attn_softmax_bf16") else None
+        )
```

Comment:
I think we should define a `attn_softmax_bf16` field in: https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/generation/configuration_utils.py
That way, we won't have to check here if it is in the generation config, and that will allow users to easily enable it in other contexts (e.g. model evaluation during training as in https://github.com/huggingface/optimum-habana/blob/4bf8a224b8838850e67ac85ca7fdbca91cc09d2d/optimum/habana/transformers/trainer_seq2seq.py#L44

Reply:
Got it. Will push patch soon.

Reply:
Done